aten/src/ATen/cuda/cub.cuh

```diff
@@ -1,4 +1,5 @@
 #pragma once
+#include <ATen/cuda/cub.h>
```

Comment:
Why do we have a cub.cuh and a cub.h? This looks confusing. Could we merge these two files?

Reply:
`cub.h` is pure C++ host code and doesn't include the actual cub library, which is quite a heavy include. So this is better for compile times.